### DISCUSSION TITLE
Unpack TypedDict into args and kwargs

### DISCUSSION THREAD

**Discussion by Kludex** [python/typing#1889] [Q&A]
Is there a way to do what the title suggests? I'm trying to improve openteletry packages and they use a lot of args and kwargs to avoid breaking their packages.

Is there a way I can do to help their type hints at least? I don't want to check if it's type checking on the signature of those functions because I want to check the rest of the body of the function.

**Comment by Viicos** [comment]
Not as of today, but there is plenty of discussion/proposals already:
- https://discuss.python.org/t/40135 (probably the most relevant) + https://github.com/python/cpython/issues/107001
- https://discuss.python.org/t/22538
- https://discuss.python.org/t/51333
- https://discuss.python.org/t/58907
- https://discuss.python.org/t/42618
